Add SOTP feature, implemented in C++

davidsaada opened the pull request.

fkjagodzinski committed:
RTOS: SysTimer: Extract RtosTimer as SysTimer

RtosTimer class introduced with tickless support in #4991 had to be
renamed because that name was already present in rtos namespace.

fkjagodzinski committed:
RTOS: SysTimer: Fix update_tick

fkjagodzinski committed:
RTOS: SysTimer: Split methods for testing

fkjagodzinski committed:
RTOS: SysTimer: Add tests

TeroJaasko committed:
nanostack-hal: add alternative critical section implementation

The nanostack hal's critical section uses a mutex for mutual exclusion,
which is nice for many use cases. But when one needs to use the critical
section from interrupts, the RTX will have a assertion failure and panic.

Add a configurable for mbed_lib, which can be used to enable a alternative
version of critical section, which uses the underlying OS primitives, which
disables the interrupts.

Note: the default behavior is not changed, one needs to override the
"nanostack-hal.critical-section-usable-from-interrupt" to have "true".

Reason for this change is that there is a need for sending events using
nanostack event queue from interrupt context, eg. from a socket callback.

dannybenor commented on features/FEATURE_SOTP/README.md:
```diff
@@ -0,0 +1,50 @@
+# SOTP
```
Need to add some documentation about how to use it (singletone, etc)

dannybenor commented on features/FEATURE_SOTP/mbed_lib.json:
```diff
@@ -0,0 +1,40 @@
+    "num_types": {
+        "macro_name": "SOTP_MAX_TYPES",
+        "value": 16,
+        "help": "Number of SOTP types"
```
Change the help to "Number of different keys allowed"

dannybenor commented on features/FEATURE_SOTP/mbed_lib.json:
```diff
@@ -0,0 +1,40 @@
+    "area_1_size": {
+        "macro_name": "SOTP_INT_FLASH_AREA_1_SIZE",
+        "value": "0x1000",
+        "help": "Size of SOTP area 1"
```
Size in bytes of...

dannybenor commented on features/FEATURE_SOTP/source/sotp.cpp:
```diff
@@ -0,0 +1,1033 @@
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+
+#define TRACE_GROUP                     "sotp"
```
Not needed anymore if we do not use mbedtrace

dannybenor commented on features/FEATURE_SOTP/source/sotp.cpp:
```diff
@@ -0,0 +1,1033 @@
+    return crc;
+}
+
+#if 0
```
Please remove

paul-szczepanek-arm commented on features/FEATURE_SOTP/source/sotp.h:
```diff
@@ -0,0 +1,297 @@
+/**
+ * @brief Returns one item of data programmed on Flash, given type.
+ *
+ * @param [in] type
```
-please explain what type means, I have no clue, maybe this should be an enum?-

I just figured it out, type is the handle/key to data
maybe type is not a very self-explanatory term for this

theotherjimmy committed:
Work around Arm Compiler 6 stdvector perfomance cliff

Commit:
Add optional tracing to sleep manager lock/unlock

Add tracing output to console to track when drivers lock and unlock deep
sleep. Tracing output is enabled by configuring the
'SLEEP_PROFILING_ENABLED' at compile time.

- Wrapped sleep_manager_lock/sleep_manager_unlock in a macro to
conditionally call tracing functions when 'SLEEP_PROFILING_ENABLED' is
set.
- Define a global structure to track driver names and how many locks
they hold in the sleep manager.

Commit:
Initial cellular feature

Commit:
Updated cellular

Commit:
Smoke test changes with easy-connect

Commit:
Changed to use platform baudrate to uart.

Commit:
Unittests added

Commit:
renamed mbed_mux to cellular_mux

Commit:
working example?

Commit:
Fixed to call callback when connected...

Commit:
Added cellular stack API

Commit:
Changed register to accept only roaming and homenetwork as registered success.

Commit:
Updated cellular doxygen

Commit:
Removed dead code from athandler.

Commit:
more unittests added

Commit:
Update AT_CellularPower.cpp

Commit:
Updated license texts.

Commit:
Removed ASSERT from mux.

Commit:
Changed requested stack for new pdp context

Commit:
Change cellular socket port 0 to dynamic range

Commit:
Removed mbed.h

Commit:
Fixed doxygen warning.

Commit:
Smaller thread stack size for dispatcher thread to save memory.

Commit:
Fixed IAR 7.8 compile

Commit:
PR review fixed: removed dead code, removed magic numbers.

Commit:
Device info buffer moved from stack to static

daid committed:
Add support for STEVAL-3DP001V1 board, which has an STM32F401VE chip. This support is based on the NUCLEO-F401RE board. Which has the same amount of flash/ram but less pins available on the chip.

Commit:
- Removed extra whitespaces from AT prefix definitions. The format is modem specific and AT handler is handling whitespaces in case they are present
- Reverted the "smaller thread stack size for dispatcher thread". 1KB is not enough, was causing CMSIS-RTOS error: Stack underflow.

Commit:
Connection status callback added.

Commit:
Deactivate PDP context on disconnect when using AT stack.

Commit:
PR review fix:
- bind behaviour to not change existing port or address if the requested ones are 0
- removed temporary array setting to zero

Commit:
Removed unnecessary memsets of ATHandler's receving buffer

Commit:
Added cellular greentea tests

theotherjimmy committed:
Correct gcc m33 floating point handling

Commit:
Changed include brackets to double quotes

Commit:
Enable connection status callback with EasyCellularConnection

Commit:
Pull request feedback fix: user_data_tx accepts arbitary size

Commit:
review fixes to athandler: changing int16/8 to int's and some minor issues.

Commit:
Added static const to AT_CellularNetwork.cpp

Commit:
Pull request feedback fix: add poll support to multiplexer

Commit:
Added const in AT_CellularNetwork::set_registration

Commit:
Doxygen warnings fixes.

Commit:
doxygen fixes.

Commit:
Cellular logging to use mbed_trace

Commit:
Doxygen update, missing class definitions from AT_ classes.

Commit:
Removed unused include in QUECTEL_BC95_CellularPower.cpp.

Commit:
Changed UBLOX C027 to LISA-U

Commit:
Changed APN initialization from CellularNetwork to application

Commit:
Cellular stack listen and accept to return UNSUPPORTED

Commit:
- BC95 socket creation bug fix
- Cellular information reading stopping on comma fixed

Commit:
Logging compile fix

jarvte committed:
review and dynamic alloc (#3)

* Review fixes. Changed apn, username and password to be dynamically allocated.

Commit:
Added cellular readme

Commit:
IOTCELL-553 fixed

Unittests reports an error as there is no coverage file created for cellular_mux.
This is now fixed by running an init function.

theotherjimmy committed:
Correct mbed export with multiple --source params

theotherjimmy committed:
Specify --build in mbed export: consistency with compile

Commit:
Fixed SMS send in pdu mode could give wrong size as return value.

Commit:
Fixed travis-ci.

Commit:
Added deleting of feature/cellular when testing a build without RTOS in .travis.yml

Commit:
Mem_str comparation was too short

Commit:
Fixed SMS receive in PDU: certain length sms last char was missing.

Commit:
Doxygen update: removed documentation from inherited classes as INHERTIC_DOCS is enabled in doxygen config. Reduces copy-paste and eases maintenance.

Commit:
Don't allow zero length apn, password or username.

Commit:
Fixed bug when copying apn, username and password.

Commit:
- PDP context creation fixes for stack type handling
- Delete created contexts that do not succed to activate

Commit:
Rename SLEEP_PROFILING_ENABLED to MBED_SLEEP_STATS_ENABLED

geky committed:
littlefs: Adopted new test configurations and status

- Now tests LFS_NO_INTRINSICS
- Also updates status with build results

Commit:
APN lookup from database support.

Commit:
review changes: take define MBED_CONF_PPP_CELL_IFACE_APN_LOOKUP in to use.

Commit:
Remove '_PACKED' keyword from sleep statistic structure

Commit:
Updated table to be const and small fixes

Commit:
Fix a bug using ELF as output extension

When overriding the OUTPUT_EXT variable in compilation configuration,
bugs appear:

* The bin variable is not None as the 'elf' string is not interned.
* When that is fixed, the function returns None instead of returning the
path of the ELF file.

Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

geky committed:
tools: Removed outdated (and broken) toolchain initialization test

Also moved the access of build_dir into condition on secure in ARMC6

per theotherjimmy

Commit:
Added modem_debug_on to cellular device

Commit:
Added use_apn_lookup in mbed_lib.json

Commit:
Remove newline printf from sleep manager

Commit:
Rename SLEEP_STATS_ENABLED to SLEEP_TRACING_ENABLED

paul-szczepanek-arm committed:
remove virtual as per review request

paul-szczepanek-arm committed:
removed using statements from header

pan- committed:
BLE: remove host testing from sm pr

pan- committed:
Merge branch 'master' of https://github.com/ARMmbed/mbed-os into pr-sm

pan- committed:
BLE: remove draft of nordic pal security manager.

bulislaw committed:
Add system_reset() function to Mbed OS

bulislaw committed:
Add tests for system_reset()

0xc0170 committed:
pull request: add required info

Please use these 2 sections for describing a pull request. They should be part
of every pull request.

The type specifies what is expected from the pull request and when it can be
released. For instance a feature pull request can't be expected to go to the
patch release.

Important: do not mix pull request types !

Changed also the heading type, to make it smaller.

paul-szczepanek-arm committed:
addressing PR review: added documentation, moved code and renamed vars

paul-szczepanek-arm committed:
Merge branch 'master' of https://github.com/paul-szczepanek-arm/mbed-os

paul-szczepanek-arm committed:
header guards renamed

cmonr committed:
Merge pull request #6166 from u-blox/apn_default

Make APN lookup the default behaviour for PPPCellularInterface

cmonr committed:
Merge pull request #6167 from hug-dev/cm3ds-bug-fix

CM3DS Maintenance Pull Request: Bug fixes (1/4)

cmonr committed:
Merge pull request #6172 from adustm/l4_linkergcc

Use SRAM2 32Kbytes on STM32L475 / L476 and L486 devices

cmonr committed:
Merge pull request #6217 from jeromecoutant/PR_RTC

STM32 RTC : update free function

cmonr committed:
Merge pull request #6189 from codeauroraforum/Update_LPC546XX_SDK2_3

LPC546XX: Update SDK driver to version 2.3

cmonr committed:
Merge pull request #6218 from geky/g-fix-travis-python-versions-2

Travis: Tear out python3 testing to get CI working again

cmonr committed:
Merge pull request #6229 from hug-dev/fix-elf-output-extension

Fix a bug using ELF as output extension

paul-szczepanek-arm committed:
link key distribution decision made in generic security manager

k-stachowiak committed:
Update Mbed TLS to version 2.7.1

k-stachowiak committed:
Adjust partner code for MD HW acceleration to new MD API

paul-szczepanek-arm committed:
fix setting default values which were overriding even when unset

Commit:
Unittests fixed + readme updated

- Unittests were broken and are now fixed
- README.md is updated to have small instructiopn how to run unittests

Commit:
Added mux-enabled flag in cellular mbed_lib.json

paul-szczepanek-arm committed:
pass in information about the mitm and sc quality of the ltk to the pal

Commit:
mcu_lpc546xx: move "release_versions" to children

Commit:
lpc546xx and lpc54114: fix clock

Commit:
Changed state machine class name and implementation.

paul-szczepanek-arm committed:
missing documentation, removed privacy call from api

adbridge committed:
Merge pull request #6209 from pan-/improve-sleep-header

Platform: Improve mbed_sleep.h

adbridge committed:
Merge pull request #6214 from tkaman/m33_build

Build issue for M33 core

adbridge committed:
Merge pull request #6227 from OpenNuvoton/nuvoton_compatible_pinName

Nuvoton: Add I2C & Serial Pin Name

adbridge committed:
Merge pull request #6225 from studavekar/measure-drift-longer

Flash test: fix the time over a longer range

Commit:
Typedef OnboardCellularInterface to EasyCellularConnection

k-stachowiak committed:
Correct typos and code alignment

paul-szczepanek-arm committed:
documentation overview for securitymanager

paul-szczepanek-arm committed:
fixed initiator distribution bug and setting LINK distribution field based on SC

cmonr committed:
Merge pull request #5911 from deepikabhavnani/common_crc

CRC class implementation

theotherjimmy committed:
Improve json schema validation errors

bulislaw committed:
Remove conflictinng system_reset() from Atmel drivers

cmonr committed:
Merge pull request #5548 from fkjagodzinski/test-systimer

Tests for SysTimer (the idle loop timer for tickless mode)

Commit:
Copy edit nvstore.h

Copy edit for typos and minor spelling nits.

Commit:
Edit README.md

Edit README with changes from docs site and minor formatting changes.

geky committed:
littlefs: Updated error codes in tests to match version update

Commit:
Copy edit README.md

Copy edit file for active voice and consistent capitalization.

Commit:
Copy edit OnboardCellularInterface.h

Copy edit file for active voice and removal of Latin abbreviations.

Commit:
Copy edit cellular_mux.h

Copy edit, mostly for spelling.

Commit:
Copy edit CellularUtil.h

Copy edit file for consistent capitalization and correct which/that.

Commit:
Copy edit AT_CellularStack.h

Copy edit for American English.

Commit:
Copy edit AT_CellularSMS.h

Copy edit file for grammar.

Commit:
Copy edit AT_CellularDevice.h

Copy edit for consistent phrasing across docs.

Commit:
Copy edit AT_CellularBase.h

Copy edit file for clarified phrasing.

Commit:
Copy edit ATHandler.h

Copy edit for clarity, grammar and spelling.

Commit:
Fixed unit tests.

Commit:
Fixed EasyCellularConnection to have debug flag

Commit:
Changed std::snprintf to sprintf in cellular ATHandler.cpp

adbridge committed:
Merge pull request #6240 from theotherjimmy/better-config-errors

                               Improve invalid configuration error messages

adbridge committed:
Merge pull request #6231 from 0xc0170/fix_pr_template_required

pull request: add required info

pan- committed:
BLE: Accept pairing request when the user authorization is not required.

pan- committed:
BLE: Implement CordioSM get_secure_connection_support.

adbridge committed:
Merge pull request #5993 from theotherjimmy/fix-vector-pref-clif

Work around Arm Compiler 6 stdvector perfomance cliff

pan- committed:
BLE: Fix CordioPalSM set_io_capability

pan- committed:
Cordio: Update stack binaries.

New binaries contains a fix in configuration handling.

paul-szczepanek-arm committed:
create encryption result event even when nothing changes

paul-szczepanek-arm committed:
Merge branch 'master' of https://github.com/paul-szczepanek-arm/mbed-os

paul-szczepanek-arm committed:
removed redundant check

we already checked the current encryption and it is encrypted

paul-szczepanek-arm committed:
fix not setting encryption state in generic

Commit:
Changed cellular device as not static

paul-szczepanek-arm committed:
fix not setting the mitm performed state correctly

pan- committed:
BLE: Return error when security deascalation is requested.

pan- committed:
BLE: Remove useless pal sm API

pan- committed:
Cordio pal SM: comment appropriately privacy related functions.

pan- committed:
BLE - Cordio PAL SM: Fix force secure connection.

pan- committed:
BLE - Cordio PAL SM: implement set_authentication_timeout

pan- committed:
BLE - Cordio PAL SM: comment get_authentication_timeout

RobMeades committed:
Call i2c_init() before i2c_frequency() (since i2c_frequency() requires the object to be set up) and zero the I2C object before giving it to i2c_init() as i2_init() has to set up a default frequency for its timeouts to work and it will only do this if the hz field in the object was zero to begin with (otherwise you end up with a random but usually very large indeed timeout).

pan- committed:
BLE - Cordio PAL SM: Choose correct security level in set_ltk

pan- committed:
BLE - Cordio PAL SM: Select correct security level in enable_encryption

pan- committed:
BLE - Cordio PAL SM: Implement send_keypress_notification

pan- committed:
BLE - Cordio PAL Sm: generate random passkey displayed.

pan- committed:
BLE - Cordio PAL SM: Forward local keys

pan- committed:
BLE - Cordio PAL SM: Select correct on_ltk_request overload.

pan- committed:
CBLE - Cordio PAL SM: Implement slave security request notification

pan- committed:
BLE - Cordio PAL SM: Implement Keypress notification

pan- committed:
BLE - Cordio - PAL SM: refine comments and formating.

Commit:
Fixed PasskeyAscii constructor

paul-szczepanek-arm committed:
added callback on encryption request even if lower and restructured for clarity

paul-szczepanek-arm committed:
Merge branch 'master' of https://github.com/paul-szczepanek-arm/mbed-os

paul-szczepanek-arm committed:
removed get encryption key size from API

pan- committed:
BLE: Remove Nordic Pal SM draft.

pan- committed:
Merge branch 'master' of https://github.com/paul-szczepanek-arm/mbed-os into pr-sm

pan- committed:
BLE: Resolve issue with address type in advertising report.

adbridge committed:
Merge pull request #5950 from theotherjimmy/header-config

Reserve and Render header in managed BL mode

cmonr committed:
Merge pull request #6230 from bulislaw/system_reset

Add system_reset call

paul-szczepanek-arm committed:
removed redundant event for SC OOB request, fixed comments

Commit:
Split find/insert sleep tracker function into two seperate functions.

Throw an error when trying to unlock driver that was not previously locked.

Commit:
Return pointer to sleep tracker array element instead of an index

Commit:
Fix merge conflict mishap

paul-szczepanek-arm committed:
no need to ask for encryption after pairing, fold deref under the null check

paul-szczepanek-arm committed:
Merge https://github.com/ARMmbed/mbed-os

paul-szczepanek-arm committed:
more detailed comment for confirmation request

paul-szczepanek-arm committed:
added one more sequence diagram

cmonr committed:
Merge pull request #6179 from geky/lfs-update

littlefs: Update to version 1.3

paul-szczepanek-arm committed:
security manager overview doc fixes

Commit:
Addressed Vincent's comments

bulislaw committed:
Fix review comments and merge issues

cmonr committed:
Merge pull request #5900 from davidsaada/david_sotp_cpp

Add NVStore (A.K.A SOTP) feature

Commit:
Removed mux.

Commit:
Fixed unit test after removing mux.

Commit:
Removed mux comments from code.

deepakvenugopal committed:
Squashed 'features/nanostack/FEATURE_NANOSTACK/coap-service/' changes from f6281ed..f40072f

f40072f Merge pull request #95 from ARMmbed/revert_interface_id_check
4b27d02 Revert "Add check for interface when receiving CoAP request (#92)"
6fd5003 Adjust request message validity lifetime (#93)

git-subtree-dir: features/nanostack/FEATURE_NANOSTACK/coap-service
git-subtree-split: f40072fe5159afacc733078375e587583ede08a8

deepakvenugopal committed:
Merge commit '41d7448660fb20d72ab17615e23c9e64005934e3'

* commit '41d7448660fb20d72ab17615e23c9e64005934e3':
  Squashed 'features/nanostack/FEATURE_NANOSTACK/coap-service/' changes from f6281ed..f40072f

Commit:
Updated readme document

adbridge committed:
Merge pull request #6188 from paul-szczepanek-arm/master

BLE: Security Manager

adbridge committed:
Merge pull request #6210 from k-stachowiak/mbed-tls-2.7.1-update

Update Mbed TLS to version 2.7.1

adbridge committed:
Merge pull request #6142 from scartmell-arm/feature-deep-sleep-tracing

Add optional tracing to sleep manager lock/unlock

adbridge committed:
Merge pull request #6251 from deepakvenugopal/master

Fixed Thread Commissioning issue in CoAP service

Commit:
Allow build profiles to omit language entries

adbridge committed:
Merge pull request #6082 from AriParkkila/master

Mbed OS cellular connectivity

Commit:
Clarify intention between comparison to zero and the shift operation

Building with (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 gives this warning

../events/equeue/equeue.c: In function 'equeue_incid':
../events/equeue/equeue.c:40:17: warning: '<<' in boolean context, did you mean '<' ? [-Wint-in-bool-context]
     if (!(e->id << q->npw2)) {

Commit:
Cellular tests: nsapi_socket_t errors fixed in unittests

cmonr committed:
Merge pull request #6246 from u-blox/i2c_fix

C030 platform I2C initialisation fix

cmonr committed:
Merge pull request #6228 from OpenNuvoton/nuvoton_1timer_ticker

Nuvoton: Rework us_ticker/lp_ticker with one H/W timer

Commit:
Flip order of comparison

Commit:
NRF5: Customize STDIO pins from config system

cmonr committed:
Modified underflow error text to print overflow instead.
End users are more familiar with handling overflow errors, even if underflow may be technically correct

SenRamakri committed:
Fix linefeed issues in crash dump output

SenRamakri committed:
Add mode and privilege info to crash dump

cmonr committed:
Merge pull request #6143 from theotherjimmy/multi-source-export

Correct mbed export with multiple --source params

Commit:
NVStore: Enclose header file with DOXYGEN_ONLY option as well.

cmonr committed:
Merge pull request #6275 from marcuschangarm/fix-uart-init

NRF52: Customize STDIO pins from config system

cmonr committed:
Merge pull request #6146 from theotherjimmy/fix-m33-fp

Correct gcc m33 floating point handling

Commit:
Incorporating review comments

theotherjimmy committed:
Print array indexes correctly in schema errors

theotherjimmy committed:
Allow _ and () in identifiers in macros

jdewinkel committed:
Small typo fixes in readme.md files

jiangkun1994 committed:
fixes a typo in README.md

jiangkun1994 committed:
fixes some typos in README.md

jiangkun1994 committed:
fixes a typo in README.md

cmonr committed:
Merge pull request #6173 from cmonr/fix-underflow-error-msg

Modifies error text reported from RTX from 'underflow' to 'overflow'

cmonr committed:
Merge pull request #6268 from AriParkkila/master

Fixes cellular unit tests.

cmonr committed:
Merge pull request #6202 from maximmbed/fix-max32625mbed

Fix max32625mbed

cmonr committed:
Merge pull request #6102 from Ultimaker/master

Add support for STEVAL-3DP001V1 board

cmonr committed:
Merge pull request #6235 from gorazdko/lpc546xx-fixes

mcu_lpc546xx: move "release_versions" to children and fix clock

cmonr committed:
Merge pull request #6263 from pauluap/equeue_gcc7_c++x11_warning

Clarify intention between comparison to zero and the shift operation

cmonr committed:
Merge pull request #6285 from theotherjimmy/fix-6284

Improve json schema and schema errors

Commit:
Fixed Cellular unit tests by adding one missing stub after changes in mbed-os.

Commit:
Cellular tests: fixed cellular greentea tests by adding timeout as it may take time to register, attach and connect to network.

0xc0170 committed:
Merge pull request #6286 from delftswa2018/feature-typo-fixes

Small typo fixes in readme.md files

0xc0170 committed:
Merge pull request #6280 from davidsaada/david_nvstore_doxygen_only

NVStore: Enclose header file with DOXYGEN_ONLY option as well.

0xc0170 committed:
Merge pull request #6271 from lfdebrux/profile-merging

Allow build profiles to omit language entries

0xc0170 committed:
Merge pull request #6226 from M-ichae-l/mbed-os_ci-test-fixed_rtl8195am

rtl8195am: fix ARM section error

0xc0170 committed:
Merge pull request #6171 from anttiylitokola/master

Update mbed-coap to version 4.3.0

0xc0170 committed:
Merge pull request #6164 from ashok-rao/br-BL652

Adding Laird BL652 as new target

theotherjimmy committed:
Merge pull request #6257 from SenRamakri/sen_FaultHandlerFixes

Fix for Crash dump formatting issues and adding more info to crash dump

geky committed:
fatfs: Revert "Update error code mapping"

This reverts commits:
b40ff8af15914e2b7aafe091997bc48872879ebd
b548fd8a39381ab20dfc3198444e0b4da9e80179

Commit:
cellular doxygen updates

0xc0170 committed:
MIMXRT: define PullUp default value

This target defines few PullUp values, one should be defined to be PullUp that
an application can use. We use the same value as PullDefault

cmonr committed:
Merge pull request #6323 from geky/revert-fat-errors

fatfs: Revert "Update error code mapping"

cmonr committed:
Merge pull request #6301 from jarvte/fixing_cellular_gt

Cellular tests: fixed cellular greentea tests by adding timeout as it…

cmonr committed:
Merge pull request #6327 from TeemuKultala/ doxygen_updates

cellular doxygen updates

cmonr committed:
Merge pull request #6328 from 0xc0170/fix_mimxrt_pullup

MIMXRT: define PullUp default value

cmonr committed:
Add lorawan and nvstore examples for testing.

0xc0170 committed:
Merge pull request #6290 from AriParkkila/fix_cellular_unit_tests

Cellular tests: fix power mgmt stub

cmonr committed:
Disabled exporters for lorawan example

cmonr committed:
Actually disabled exporters for lorawan example

cmonr committed:
Merge pull request #6341 from cmonr/add-5.8-examples

Add lorawan and nvstore examples for testing.

Commit:
NVStore: add comments in header file for Doxygen formatting sake.
